### PR TITLE
ACLs for financial types

### PR DESCRIPTION
**Pull request by JoeMurray**
This is a preliminary PR that is not intended to be merged but just for preliminary code review by core team. It will be updated with fixes for reports for various edge cases.

### CODE REVIEW

File: CRM/Financial/BAO/FinancialType.php [hunk 1]
```diff
@@ -181,11 +181,102 @@ public static function getIncomeFinancialType() {
...
+    foreach ($financialTypes as $id => $type) {
+      $permissions['add contributions of type ' . $type] = $prefix . ts('add contributions of type ' . $type);
+      $permissions['view contributions of type ' . $type] = $prefix . ts('view contributions of type ' . $type);
+      $permissions['edit contributions of type ' . $type] = $prefix . ts('edit contributions of type ' . $type);
+      $permissions['delete contributions of type ' . $type] = $prefix . ts('delete contributions of type ' . $type);
+    }
+    $permissions['administer CiviCRM Financial Types'] = $prefix . ts('administer CiviCRM Financial Types');
+  }
+
+  public static function getAvailableFinancialTypes(&$financialTypes = NULL, $action = 'view') {
```
Comment (JoeMurray) [inline CRM/Financial/BAO/FinancialType.php]: $action is not referenced in function, so you might want to drop it.

Comment (Edzelopez) [inline CRM/Financial/BAO/FinancialType.php]: Joe, $action is indeed referenced in the function. By default, it is 'view' but in certain instances, we can change it to check for various CRUD permissions by changing $action while invoking.

**Comment by pradpnayak** [comment]
closing this against https://github.com/civicrm/civicrm-core/pull/6431

**Comment by eileenmcnaughton** [comment]
Closing per @pradpnayak

**Pull request closed without merging**